lib/cocoapods/validator.rb

```diff
@@ -470,7 +470,7 @@ def install_pod
           next unless (native_target = pod_target.native_target)
           native_target.build_configuration_list.build_configurations.each do |build_configuration|
             (build_configuration.build_settings['OTHER_CFLAGS'] ||= '$(inherited)') << ' -Wincomplete-umbrella'
-            build_configuration.build_settings['SWIFT_VERSION'] = swift_version if pod_target.uses_swift?
+            build_configuration.build_settings['SWIFT_VERSION'] = swift_version if pod_target.uses_swift? && !swift_version.nil?
```

Comment:
is there a test that covers the `!swift_version.nil?` part? If not can you please add one?

Reply:
i'm not a ruby developer, i will try my best to write unit tests

Reply:
not a problem! There are plenty of examples around the `*_spec.rb` for the class you are writing a test for you can cargo cult from.

Reply:
i'm trying to figured out how to run unit tests in ruby. reading this tutorial here: https://code.tutsplus.com/tutorials/ruby-for-newbies-testing-with-rspec--net-21297
but i'm getting this error:

```
/Users/victor/.rvm/gems/ruby-2.0.0-p648/gems/rspec-core-3.6.0/lib/rspec/core/formatters.rb:206:in `built_in_formatter': cannot load such file -- rspec/core/formatters/progress_formatter (LoadError)
```

googling it to figured out how to fix it. if know the solution, let me know please.

Reply:
nvm... i figured how to run `bundle exec bacon ...`

Reply:
you can also run `bundle exec rake spec` or `bundle exec rake spec:unit` for only unit tests

Reply:
@heyzooi can you provide a case where the part `!swift_version.nil?` can be true?